src/core/utils.js

```diff
@@ -354,9 +355,14 @@ export function isCompatibleWithUserAgent({
     // confused by this as tofumatt was.
     // See: https://github.com/mozilla/addons-frontend/issues/2074#issuecomment-286983423
     if (maxVersion && mozCompare(browser.version, maxVersion) === 1) {
-      _log.info(oneLine`maxVersion ${maxVersion} for add-on lower than browser
-        version ${browser.version}, but add-on still marked as compatible
-        because we largely ignore maxVersion. See:
+      // See: https://github.com/mozilla/addons-server/issues/5277
+      if (addon.current_version.is_strict_compatibility_enabled) {
+        return { compatible: false, reason: INCOMPATIBLE_LEGACY_EXTENSION };
```

Comment:
This looks like the right thing to do but this message should be more specific to strict compatibility. We already have `INCOMPATIBLE_UNDER_MIN_VERSION` so I would suggest `INCOMPATIBLE_OVER_MAX_VERSION` which is to say the user's Firefox version is higher than the max version allowed by the extension.

Also, keep in mind that any add-on can opt into strict compatibility checks, not just "legacy" add-ons. A web extension add-on can say that it's using an experimental API which will be removed in Firefox 59, for example. Furthermore, beta add-on releases typically say they are only compatible with the current version of Firefox to dissuade ongoing usage.